pandas/core/indexes/base.py

```diff
@@ -542,9 +541,10 @@ def __new__(
                     # 10697
                     from pandas.core.indexes.multi import MultiIndex
 
-                    return MultiIndex.from_tuples(
-                        data, names=name or kwargs.get("names")
-                    )
+                    if all(data):
```

Comment:
instead of this you should modify `from_tuples` itself

Reply:
Will update it.

Reply:
Have update it.